libs/application/templates/children-residence-change/src/dataProviders/MockNationalRegistryProvider.ts

```diff
@@ -0,0 +1,84 @@
+import {
+  Application,
+  BasicDataProvider,
+  FailedDataProviderResult,
+  SuccessfulDataProviderResult,
+} from '@island.is/application/core'
+
+import { PersonResidenceChange } from '@island.is/application/templates/children-residence-change'
+import { NationalRegistry, DataProviderTypes, CRCApplication } from '../types'
+
+export class MockNationalRegistryProvider extends BasicDataProvider {
+  readonly type = DataProviderTypes.MOCK_NationalRegistry
+
+  async provide(application: Application): Promise<NationalRegistry> {
+    const query = `
+      query NationalRegistryUserQuery {
+        nationalRegistryUser {
+          nationalId
+          fullName
+          address {
+            code
+            postalCode
+            city
+            streetAddress
+            lastUpdated
+          }
+        }
+      }
+    `
+    const crcApplication = (application as unknown) as CRCApplication
+    const {
+      answers: {
+        mockData: { parents, children },
+      },
+    } = crcApplication
+    if (!children) {
+      throw new Error('Engin börn fundust í þinni forsjá')
+    }
+    const childrenArray: NationalRegistry['children'] = []
+    children?.map((child) => {
+      childrenArray.push({
+        ...child,
+        livesWithApplicant: child?.livesWithApplicant?.includes('yes') || false,
+        otherParent: parents[child.otherParent],
+      })
+    })
```

Comment:
Instead of setting the `childrenArray` to an empty array and pushing to that array we could do this: 
```    const childrenArray: NationalRegistry['children'] = children.map(
      (child) => {
        return {
          ...child,
          livesWithApplicant:
            child.livesWithApplicant.includes('yes') || false,
          otherParent: parents[child.otherParent],
        }
      },
    )